crates/uv/src/commands/build.rs

```diff
@@ -0,0 +1,338 @@
+                .map(RequiresPython::specifiers)
+                .map(|specifiers| {
+                    PythonRequest::Version(VersionRequest::Range(specifiers.clone()))
+                });
+        }
+    }
+
+    // Locate the Python interpreter to use in the environment.
+    let interpreter = PythonInstallation::find_or_download(
+        interpreter_request.as_ref(),
+        EnvironmentPreference::Any,
+        python_preference,
+        python_downloads,
+        &client_builder,
+        cache,
+        Some(&PythonDownloadReporter::single(printer)),
+    )
+    .await?
+    .into_interpreter();
+
+    // Add all authenticated sources to the cache.
+    for url in index_locations.urls() {
+        store_credentials_from_url(url);
+    }
+
+    // Initialize the registry client.
+    let client = RegistryClientBuilder::new(cache.clone())
+        .native_tls(native_tls)
+        .connectivity(connectivity)
+        .index_urls(index_locations.index_urls())
+        .index_strategy(index_strategy)
+        .keyring(keyring_provider)
+        .allow_insecure_host(allow_insecure_host.to_vec())
+        .markers(interpreter.markers())
+        .platform(interpreter.platform())
+        .build();
+
+    // Determine whether to enable build isolation.
+    let environment;
+    let build_isolation = if no_build_isolation {
+        environment = PythonEnvironment::from_interpreter(interpreter.clone());
+        BuildIsolation::Shared(&environment)
+    } else if no_build_isolation_package.is_empty() {
+        BuildIsolation::Isolated
+    } else {
+        environment = PythonEnvironment::from_interpreter(interpreter.clone());
+        BuildIsolation::SharedPackage(&environment, no_build_isolation_package)
+    };
+
+    // TODO(charlie): These are all default values. We should consider whether we want to make them
+    // optional on the downstream APIs.
+    let build_constraints = [];
+    let hasher = HashStrategy::None;
+
+    // Resolve the flat indexes from `--find-links`.
+    let flat_index = {
+        let client = FlatIndexClient::new(&client, cache);
+        let entries = client.fetch(index_locations.flat_index()).await?;
+        FlatIndex::from_entries(entries, None, &hasher, build_options)
+    };
+
+    // Initialize any shared state.
+    let state = SharedState::default();
+
+    // Create a build dispatch.
+    let build_dispatch = BuildDispatch::new(
+        &client,
+        cache,
+        &build_constraints,
+        &interpreter,
+        index_locations,
+        &flat_index,
+        &state.index,
+        &state.git,
+        &state.in_flight,
+        index_strategy,
+        config_setting,
+        build_isolation,
+        link_mode,
+        build_options,
+        exclude_newer,
+        sources,
+        concurrency,
+    );
+
+    // Determine the build plan from the command-line arguments.
+    let path = path.unwrap_or_else(|| &*CWD);
+    let output_dir = path.join("dist");
```

Comment:
We should make this configurable and add a `.gitignore` by default (similar to cache dirs)

Reply:
Will do RE `outdir`. Not sure on `.gitignore `though...